File: SAE_2.02_Java/src/main/eraser/EraserTrois.java
package main.eraser;

import java.util.ArrayList;
import java.util.Arrays;

public class EraserTrois {
	/**
	 * Méthode par copy de chaine de caractère pour supprétion
	 * @param req
	 * @return
	 */
	public static String start(String req) {
		ArrayList<String> res = new ArrayList<String>(Arrays.asList(req.split("")));
		if (res.get(0).equals(" ") && !res.get(1).equals(" ")) {
			res.remove(0);
		}
		for (int i = 0; i < res.size() - 1; i++) {
			if (res.get(i).equals(" ") && !res.get(i+1).equals(" ") && !res.get(i-1).equals(" ")) {
				res.remove(i);
			}
		}
		if (res.get(res.size()-1).equals(" ") && !res.get(res.size()-2).equals(" ")) {
			res.remove(res.size()-1);
		}
		return String.join("", res);
	}
}

File: SAE_2.02_Python/eraserTrois.py
from copy import copy
from ntpath import join


def start(req):
    res = list(req)
    i = 1
    test = True
    if (res[0] == ' ' and res[1] != ' '):
        del res[0]

    while (test):
        if (i >= len(res)-1):
            break
        if (res[i] == ' ' and res[i+1] != ' ' and res[i-1] != ' '):
            del res[i]
        i += 1

    if res[len(res)-1] == ' ' and res[len(res)-2] != ' ':
        del res[len(res)-1]
    return ''.join(res)
